runtime/str.go

```diff
@@ -717,6 +757,23 @@ func adjustIndex(i, l int) int {
 	return i
 }
 
+func adjustBounds(f *Frame, o *Object, l int, upperBound bool) (int, *BaseException) {
```

Comment:
I'm worried about adding this function because it's not obvious when this should be used vs. adjustIndex. Generally I think the Index/IndexInt/adjustIndex functions are confusing and need to be cleaned up. I would prefer to do this logic inline in strFind rather than adding another function to the mix. You could reuse adjustIndex by doing it as follows:

```
if argc >= 3 {
  start, raised = IndexInt(f, args[2])
  if raised != nil {
    return nil, raised
  }
}
if argc == 4 {
	end, raised = IndexInt(f, args[3])
	if raised != nil {
		return nil, raised
	}
}
if start > end {
	return NewInt(-1).ToObject(), nil
}
start, end = adjustIndex(start, l), adjustIndex(end, l)
```

I may be missing some edge cases here, but the nice thing is that we don't need to add any new helper functions and the semantics are obvious from reading the code.